### DISCUSSION TITLE
Variant chart

### DISCUSSION THREAD

**Discussion by AdamLeaghj** [mtex-toolbox/mtex#1651] [Ask Anything] [answered]
Hello,

I used `histogram(job.variantId,parentGrains,'facecolor','r')` to draw my p2c variant. How to change the y-axis to area fraction? I couldn't find the class job.childGrains in my workspace. Thanks.

Here below is my current variant chart.
![image](https://user-images.githubusercontent.com/118141541/201551501-944ba27c-9224-49df-ac03-13c8ed191ae2.png)

**Comment by AdamLeaghj** [comment]
Please let me know if anyone has a solution for this. Thank you.

**Comment by ralfHielscher** [comment]
Hi Adam,

you can do 
```matlab 
histogram(job.transformedGrains,job.transformedGrains.variantId)
```
Ralf.

**Reply by AdamLeaghj** [reply]
Thanks Ralf!

**Reply by AdamLeaghj** [reply]
Hi Ralf, 

Just a follow-up question: I just used your code but why the result returned with only 14 variants? Thanks.

![image](https://user-images.githubusercontent.com/118141541/203416782-1461fc7f-a519-426c-8a7f-96a9ccd66477.png)

**Comment by ralfHielscher** [comment, marked as answer]
Its because ```histogram``` is a generic histogram function that does not know how many bin it should make. See https://mtex-toolbox.github.io/grain2d.histogram.html.
```matlab
histogram(job.transformedGrains,job.transformedGrains.variantId,24)
```
to get the bin number right.
Ralf.

**Reply by AdamLeaghj** [reply]
Thanks Ralf! Just a follow up question: what if I want to plot this data using EBSD data instead of grain data? Thanks.

**Reply by Duanyouteng** [reply]
> Its because `histogram` is a generic histogram function that does not know how many bin it should make. See https://mtex-toolbox.github.io/grain2d.histogram.html.
> 
> ```matlab
> histogram(job.transformedGrains,job.transformedGrains.variantId,24)
> ```
> 
> to get the bin number right. Ralf.

dear Ralf：

I used the code you provided to count the variants, and the following error occurred when running.
![image](https://user-images.githubusercontent.com/123815884/215269735-65cd81c2-841d-45ae-8c1e-a7b6a444b373.png)

"出错 grain2d/subsref (line 83)
  [varargout{1:nargout}] = builtin('subsref',grains,s);"

May I ask how to solve this problem.
Looking forward to your reply
Duan.
